flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/FlinkFilterJoinRule.java

```diff
@@ -431,13 +431,16 @@ public void onMatch(RelOptRuleCall call) {
         }
 
         /** Rule configuration. */
-        public interface Config extends FlinkFilterJoinRule.Config {
-            FlinkJoinConditionPushRule.Config DEFAULT =
-                    EMPTY.withOperandSupplier(b -> b.operand(Join.class).anyInputs())
-                            .as(FlinkJoinConditionPushRule.Config.class)
-                            .withSmart(true)
-                            .withPredicate((join, joinType, exp) -> true)
-                            .as(FlinkJoinConditionPushRule.Config.class);
+        @Value.Immutable(singleton = false)
+        @Value.Style(
+                get = {"is*", "get*"}, // Detect 'get' and 'is' prefixes in accessor methods
+                init = "with*", // Builder initialization methods will have 'set' prefix
+                defaults = @Value.Immutable(copy = false))
```

Comment:
This is related to the next line 
```java
public interface FlinkFilterJoinRuleConfig extends FlinkFilterJoinRule.Config
```
where `FlinkFilterJoinRule.Config` instead of `set` methods has `Config withSmart(boolean smart);` and `Config withPredicate(Predicate predicate);`
By default it generates setters with `set` prefix. Via such configuration it's possible to say that we need setters with `with` prefix to match `FlinkFilterJoinRule.Config` methods.

Reply:
I think what confuses me here is that the comment states `will have 'set' prefix` but actually `will have 'with' prefix`. How about we simply drop the comments here and above. IMHO the code is self explanatory.

Reply:
yes... probably you are right
removed these comments